.github/workflows/checks.yaml

```diff
@@ -304,6 +304,45 @@ jobs:
           cargo build --release --target ${{matrix.target}} --features=custom-prefix
           objdump -tT target/${{matrix.target}}/release/deps/libz_rs.so | grep -q "MY_CUSTOM_PREFIX_uncompress" || (echo "symbol not found!" && exit 1)
 
+  cargo-c-dynamic-library:
+    name: cargo-c dynamic library
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+        features:
+          - ''
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@be73d7920c329f220ce78e0234b8f96b7ae60248
+        with:
+          toolchain: stable
+          targets: ${{matrix.target}}
+      - name: Install cargo-c
+        env:
+          LINK: https://github.com/lu-zero/cargo-c/releases/latest/download
+        run: |
+          curl -L "$LINK/cargo-c-x86_64-unknown-linux-musl.tar.gz" |
+          tar xz -C $HOME/.cargo/bin
+      - name: build with and test the result of cargo-c
+        working-directory: libz-rs-sys-cdylib
+        run: |
+          # build using cargo-c this time
+          cargo cinstall --offline --release --destdir=/tmp/cargo-cbuild-libz-rs
+          tree /tmp/cargo-cbuild-libz-rs
+          # verify that the SONAME  is set and includes a version
+          objdump -p target/x86_64-unknown-linux-gnu/release/libz_rs.so | awk '/SONAME/{print $2}' | grep -E 'libz_rs\.so\.[0-9]+\.[0-9]+'
```

Comment:
The original zlib uses `libz.so.1` as SONAME. I think using `libz_rs.so.0.4` as SONAME for zlib-rs is wrong. We are matching the ABI of the original zlib and when zlib-rs bumps to version 0.5, the ABI will stay the same and thus SONAME shouldn't change.

Reply:
you may set `package.metadata.capi.library.version` to address it.

Reply:
hmm. That `0.4` is the rust version, and not the version of zlib that we mirror. We're not at 1.0 yet with zlib-rs, so in the meantime, how should we handle the situation?

I think mostly this c dylib is for individuals that want to use zlib-rs in their projects; we're not ready for distributions yet (I expect that to only happen at 1.0 and later). So is there a problem in practice here?

Reply:
Using a version that doesn't have a relation to the ABI in SONAME kinda invalidates the point of having SONAME in the first place I would think. There are two main users of dylibs:

* Distros: These want SONAME to match the ABI compatibility such that they can upgrade the dylib without needing to rebuild everything when the new version is ABI compatible with the old version.
* Projects who bundle the dylib as part of their program: These don't need SONAME at all as the dylib will never be updated independently of the program that uses it.

Reply:
cool, set that to `1.3.0` now, which is the actual api version that we say we provide (though some stuff is missing currently)